source/common/grpc/google_async_client_impl.h

```diff
@@ -219,8 +221,9 @@ class GoogleAsyncStreamImpl : public RawAsyncStream,
   void sendMessageRaw(Buffer::InstancePtr&& request, bool end_stream) override;
   void closeStream() override;
   void resetStream() override;
-  // The GoogleAsyncClientImpl doesn't do Envoy watermark based flow control.
-  bool isAboveWriteBufferHighWatermark() const override { return false; }
+  bool isAboveWriteBufferHighWatermark() const override {
```

Comment:
Question: do you think it's a bit confusing to talk about high watermark here? What we really have is a per async client buffer limit; there isn't low/high watermark in general.

Reply:
Friendly ping on this one.

Reply:
I added a comment in the last push to try to clarify it, did you not see that change?  Or are you suggesting I should instead rename the virtual interface throughout the rest of Envoy because the gRPC shim doesn't use Envoy constructs?

Reply:
Not in the rest of Envoy, in https://github.com/envoyproxy/envoy/blob/master/include/envoy/grpc/async_client.h#L66. I.e. for async clients, the concept isn't necessarily watermarks, it's available client capacity.

Reply:
Sure, but it is watermarks for the other cases, it's really just gRPC that's doing its own thing. I think that any API extensions could end up with the functional end point not being watermarks - add a fake IO socket for writing directly to a socket and there's no longer a watermark buffer in that connection end class, there's a shared memory region.   I think the API consistency is more important than the implementation details but happy to let Matt tiebreak if you feel strongly about it and we can rename async client APIs to something that's not consistent with the rest of Envoy flow control but is consistent with the google-grpc client instead.

Reply:
I don't think these kinds of naming discussions are worth going getting too hung up on; either way we go there is something weird happening; we're either being inconsistent with the API or misleading. So, I'm happy to resolve for now, just ensuring my original comment was correctly understood.